composeApp/src/commonMain/kotlin/designsystem/login/CredentialInput.kt

```diff
@@ -0,0 +1,95 @@
+package designsystem.login
+
+import designsystem.colors.ButtonColor
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.material.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.dp
+import designsystem.colors.TextColor
+import dev.icerock.moko.resources.ImageResource
+import dev.icerock.moko.resources.compose.painterResource
+import instaclone.resources.MR
+
+@Composable
+fun CredentialInput(
```

Comment:
I don't understand. Is this component used for both fields e-mail, and password!?

I think it would be better we have two different components: `EmailInputFeild`, and `PasswordInputField`.

Reply:
Something like:

```kotlin
fun EmailInputField(currentValue: String, onValueChanged: (String) -> Unit, modifier) {...}

fun PasswordInputField(currentValue: String, onValueChanged: (String) -> Unit, modifier) {...}
```

Reply:
Sounds good